monai/data/utils.py

```diff
@@ -149,6 +149,48 @@ def iter_patch_slices(
         yield tuple(slice(s, s + p) for s, p in zip(position[::-1], patch_size_))
 
 
+def iter_wsi_patch_location(
+    image_size: Sequence[int],
+    patch_size: Union[Sequence[int], int],
+    downsample: float = 1.0,
+    overlap: float = 0.0,
+    start_pos: Sequence[int] = (),
+    padded: bool = False,
+):
+    """
+    Yield successive tuple of locations defining a patch of size `patch_size` from an image of size `image_size`,
+    with the relative overalpping of `overlap`. The patch is in the resolution level related to `downsample` ratio.
+    The iteration starts from position `start_pos` in the whole slide image, or starting at the origin if this isn't
+    provided.
+
+    Args:
+        image_size: dimensions of image
+        downsample: the downsample ratio the
+        patch_size: size of patches to generate slices for, 0 or None selects whole dimension
+        downsample: the relative amount of overlap for patches
+        start_pos: starting position in the image, default is 0 for each dimension
+        padded: if the image is padded so the patches can go beyond the borders. Defaults to False.
+            Note that the padding depends on the functionality of the underlying whole slide imaging reader,
+            and is not guranteed for all images.
+
+    Yields:
+        Tuple of patch location
+    """
+    ndim = len(image_size)
+    patch_size = get_valid_patch_size(image_size, patch_size)
+    start_pos = ensure_tuple_size(start_pos, ndim)
+
+    # Get the patch size at level=0
+    patch_size_0 = [p * downsample for p in patch_size]
```

Comment:
I think the user can directly input `patch_size_0` instead of `downsample` and `patch_size`?

Reply:
I see you point but the problem is that the user cares about the `patch_size` (at the resolution level she is working with) and does not necessarily know `patch_size_0`, which is the patch size at `level=0`.

Reply:
I see you point but the problem is that the user cares about the patch_size (at the resolution level she is working with) and does not necessarily know patch_size_0, which is the patch size at level=0.